routing/astar_router.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "routing/directions_engine.hpp"
+#include "routing/road_graph.hpp"
+#include "routing/router.hpp"
+#include "routing/vehicle_model.hpp"
+
+#include "indexer/index.hpp"
+
+#include "std/shared_ptr.hpp"
+#include "std/unique_ptr.hpp"
+#include "std/vector.hpp"
+
+namespace routing
+{
+class IndexGraph;
+
+class AStarRouter : public IRouter
+{
+public:
+  AStarRouter(const char * name, Index const & index, TCountryFileFn const & countryFileFn,
+              shared_ptr<VehicleModelFactory> vehicleModelFactory,
+              unique_ptr<IDirectionsEngine> directionsEngine);
+
+  // IRouter overrides:
+  string GetName() const override;
+  ResultCode CalculateRoute(m2::PointD const & startPoint, m2::PointD const & startDirection,
```

Comment:
IRouter::CalculateRoute(...) can be use to connect two any points on map with a route.
AStarRouter::CalculateRoute(...) cannot be used for this. It can only connect two points with a route within one mwm. Please write it to documentation for AStarRouter::CalculateRoute(...) and add an appropriate assert in AStarRouter::CalculateRoute(...).